core/util/runPuppet.js

```diff
@@ -72,10 +72,16 @@ async function processScenarioView (scenario, variantOrScenarioLabelSafe, scenar
 
   // --- set up console output and ready event ---
   const readyEvent = scenario.readyEvent || config.readyEvent;
-  let readyResolve, readyPromise;
+  const readyTimeout = scenario.readyTimeout || config.readyTimeout || 30000;
+  let readyResolve, readyPromise, readyTimeoutTimer;
   if (readyEvent) {
     readyPromise = new Promise(resolve => {
       readyResolve = resolve;
+      // fire the ready event after the readyTimeout
+      readyTimeoutTimer = setTimeout(() => {
+        console.error(chalk.red('*** readyEvent Timeout! ***'), scenario.url);
```

Comment:
Could we make this log more explicit?  
_ReadyEvent not detected within readyTimeout limit. (\`${readyTimeout}\` seconds)_

Reply:
Fixed.